Packs/MyXSIAMPack_content/Integrations/OKTA_SIEM_Data/OKTA_SIEM_Data.py

```diff
@@ -0,0 +1,302 @@
+        for key, value in params.items():
+            if key == 'query':
+                key = 'q'
+            query_params[key] = encode_string_results(value)
+        # if args.get('limit'):
+        #     return self._http_request(
+        #         method='GET',
+        #         url_suffix=uri,
+        #         params=query_params
+        #     )
+        return self.get_paged_results(uri, query_params)
+
+    def fetch_logs(self):
+        new_last_run = last_run = demisto.getLastRun()
+        demisto.info(json.dumps(last_run))
+        fetch_start_time = last_run.get('last_event_created_time') or self.first_fetch
+        previous_fetched_uuids = last_run.get('previous_fetched_uuids') or []
+
+        params = {
+            'limit': str(self.fetch_limit),
+            'since': fetch_start_time
+        }
+
+        sorted_events_to_save = []
+        res = self.get_logs(params)
+        if res:
+            new_events = []
+            new_events_uuids = []
+            for item in res:
+                if (item_uuid := item.get('uuid')) not in previous_fetched_uuids:
+                    new_events.append(item)
+                    new_events_uuids.append(item_uuid)
+
+            if new_events:
+                # OKTA doesn't sort events by milliseconds so we sort the list before returning.
+                sorted_events_to_save = sorted(new_events, key=lambda obj: arg_to_datetime(obj.get('published')))
+                last_event_created_time = sorted_events_to_save[-1].get('published')
+
+                if not self.should_reset_list(fetch_start_time, last_event_created_time):
+                    new_events_uuids = previous_fetched_uuids + new_events_uuids
+
+                new_last_run = {
+                    'last_event_created_time': last_event_created_time,
+                    'previous_fetched_uuids': new_events_uuids
+                }
+
+
+        return sorted_events_to_save, new_last_run
+
+
+def test_http_collector():
+    headers = {
+        'Authorization': XSIAM_API_KEY,
+        "Content-Type": "application/json"
+    }
+
+    data = [{"example1": "test", "timestamp": 1609100113039}, {"example2": [12321, 546456, 45687, 1]}]
+
+    res = requests.post(XSIAM_SERVER_URL, headers=headers,
+                        data=json.dumpps(data))
+    demisto.info(f'\n\n{res.reason}\n\n\n')
+    demisto.info(f'\n\n{res.status_code}\n\n\n')
+    return res
+
+
+def send_logs_to_xsiam(client, logs):
+    headers = {
+        "Authorization": XSIAM_API_KEY,
+        "Content-Type": "text/plain "
+    }
+    res = client._http_request(method='POST', full_url=XSIAM_SERVER_URL, headers=headers, data=logs)
+    # Note: the logs must be separated by a new line
+    # res = requests.post(url=XSIAM_SERVER_URL, headers=headers, data=logs)
+
+    return res
+
+
+def save_logs(client, logs):
+    formatted_logs = '\n'.join([json.dumps(log) for log in logs])
+    res = send_logs_to_xsiam(client, formatted_logs)
```

Comment:
`res` isn't used.
Maybe print it to the debug log with the http status (and message if it isn't 200) or success/fail?

Reply:
@tkatzir This is not a code that will be merged to master. The logic of saving the logs to XSIAM will be done elsewhere.

Reply:
Fair enough,
but for maintenance reasons I'd log all output of communications with "external systems" (including all REST calls.)